src/components/DeleteSessionDialog.vue

```diff
@@ -0,0 +1,50 @@
+<script setup>
+import { fetchApiCall } from '@/utils/api';
+const props = defineProps([ 'modelValue', 'deleteId'])
+const emit = defineEmits(['update:modelValue', 'reloadSession'])
+
+function closeDialog() { emit('update:modelValue', false)}
+
+async function confirmDeleteSession() {
+  if (props.deleteId != -1) {
+    const url = 'http://127.0.0.1:8000/api/datasessions/' + props.deleteId
+    fetchApiCall({url:url, method:'DELETE'})
```

Comment:
@capetillo's fetchApiCall can take callbacks on success/fail, so we should probably do these callbacks that way to stay consistent